rcsbapi/graphql_schema.py

```diff
@@ -1072,23 +1083,42 @@ def add_field_args(self, field_name: str, args: List[dict[str, Any]], query_args
         """Add arguments to a field, returning the fieldname and args as a formatted string.
 
         Args:
-            args (list[dict[str, Any]]): args of a field, retrieved from the GraphQL schema/FieldNode object
-            query_args (dict[str, Any]): dictionary where keys are argument name and values are user input
+            field_name (str): name of the field
+            args (list[dict[str, Any]]): argument definitions from schema
+            query_args (dict[str, Any]): full dictionary of query arguments
 
         Returns:
-            str: field name or field name with corresponding arguments
+            str: field name, potentially with arguments applied (e.g., field(arg: val))
+
+        Raises:
+            ValueError: if an invalid field argument key is found
         """
-        # Check FieldNode argument names and see if user has passed in corresponding values
+        field_args_dict = query_args.get("datalistargs", {})
+
+        # Validate all user-provided field names
+        user_field_args = field_args_dict.get(field_name, {})
+
+        for user_key in field_args_dict.keys():
+            if user_key != field_name:
+                raise ValueError(
+                    f"Invalid field argument '{user_key}' in data_list_args. "
+                    f"Allowed fields for this query are: {list(self._field_names_list)}"
+                )
+
         formatted_args = []
         for arg in args:
             arg_name = arg["name"]
-            if arg_name in query_args:
-                formatted_args.append(self._format_args(arg, query_args[arg_name]))
+            if arg_name not in user_field_args:
+                continue
+            val = user_field_args[arg_name]
+            if isinstance(val, str):
+                formatted_args.append(f'{arg_name}: "{val}"')
+            else:
+                formatted_args.append(f"{arg_name}: {val}")
 
         if formatted_args:
-            return "{}{}".format(field_name, str(tuple(formatted_args)).replace("'", ""))
-        else:
-            return field_name
+            return f"{field_name}({', '.join(formatted_args)})"
+        return field_name
```

Comment:
Can you please add some comments throughout this method to clarify what each step is doing? It's easy to get lost here...